libraries/@mattermost/rnshare/android/src/main/java/com/mattermost/rnshare/ShareWorker.kt

```diff
@@ -198,7 +199,7 @@ class ShareWorker(private val context: Context, workerParameters: WorkerParamete
                 .setSmallIcon(applicationContext.resources.getIdentifier("ic_notification", "mipmap", applicationContext.packageName))
                 .setOngoing(true)
                 .build()
-        return ForegroundInfo(1, notification)
+        return ForegroundInfo(1, notification, ServiceInfo.FOREGROUND_SERVICE_TYPE_DATA_SYNC)
```

Comment:
Looks good but do we need to check the permission at runtime just in case?

Reply:
@enahum Good question, would we need to?  Are we worried that user might disable foreground service of this type?  I have to look if that's even possible.

Reply:
Do check if needed, and if so, let's add it

Reply:
@enahum https://support.google.com/googleplay/android-developer/answer/13392821?hl=en#zippy=%2Chow-do-i-know-what-foreground-service-type-to-use%2Cwhat-is-a-foreground-service-permission-and-does-it-need-to-be-granted-by-the-user-before-starting-a-foreground-service

> When apps that target Android 14 use a foreground service, developers must declare the appropriate foreground service permission for that specific foreground service type. For example, if your app uses the "location" foreground service type, you will need to declare the [FOREGROUND_SERVICE_LOCATION](https://developer.android.com/reference/android/Manifest.permission#FOREGROUND_SERVICE_LOCATION) permission. Foreground service permissions that refer to a specific foreground service type are defined as [normal permissions](https://developer.android.com/guide/topics/permissions/overview#install-time) and are granted by default at install time. Users cannot revoke these permissions.

So I think we should be okay.